pkg/codegen/schema/schema.go

```diff
@@ -1533,7 +1533,7 @@ type DiscriminatorSpec struct {
 // DefaultSpec is the serializable form of extra information about the default value for a property.
 type DefaultSpec struct {
 	// Environment specifies a set of environment variables to probe for a default value.
-	Environment []string `json:"environment,omitempty" yaml:"environment,omitempty"`
+	Environment []string `json:"environment" yaml:"environment"`
```

Comment:
This is marked as required here https://www.pulumi.com/docs/iac/using-pulumi/pulumi-packages/schema/#default, so this seems correct, but what actually is the source of truth for this?

Reply:
I'm not actually sure what the source of truth for this is.  Maybe it's better to stay more liberal with parsing it here?

Reply:
Yeah, I am worried that this might cause issues for someone somewhere ... I think we can leave the omit empty for now.